### PR TITLE
feat(specs): define GitHub repo preferences sync — specs, ADRs, test cases (#64)

### LINKED ISSUE #64: Create a workflow to allow a user to save / read his preferences to a dedicated repository as database
A user can decide to store his station list and fuel type default to a repository he owns through a GitHub app that is granted read and write access to a single JSON file (same structure as the static file exported/imported in the app).

It must handle public and private repositories.

The user's data from the repo should remain persited to the application's indexedDb. Validation of indexedDb will work as follows:
- if indexedDb data less than a week old (7 days as a parameter `revalidate-cache-days` updatable from settings UI), read indexedDb
- else refresh indexedDb data from remote repo data
- each update triggered by user reset `revalidate-cache-days` value to current date and time

If access is revoked to GitHub App, then, the indexedDb source is preferred. A warning must appear on the app.

When the indexedDb is updated, a request to update the remote file on user's repository happen.
When updating the remote file on user's repository, its content should be read first to present the user a diff UI as implemented in issue #63.

The credentials used are GitHub's since the user's data is saved to a repository he owns.

This issue should be split into several sub-issues because of the feature's complexity and requirement for proper planning.

### PR DESCRIPTION
**Pull request by JeremieLitzler**
## Summary

- Defines business specifications, security guidelines, and test cases for issue #64 (GitHub repository as preferences sync backend)
- Documents two new architectural decisions (ADR-011, ADR-012) covering OAuth authentication and remote sync
- Creates 6 child GitHub issues (Sub-Issues A–F) with dependency links, ready for implementation

## What's in this PR

| File | Purpose |
|------|---------|
| `docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md` | 6 sub-issues with rules, testing guides (local + going live), Mermaid OAuth diagram, remote JSON schema |
| `docs/prompts/tasks/issue-64-github-repo-preferences/security-guidelines.md` | Security review output |
| `docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md` | 33 test scenarios across sub-issues A–F (Pass 1) |
| `docs/decisions/ADR-011-github-oauth-app-auth.md` | GitHub OAuth App flow with HTTP-only cookie via Netlify Functions |
| `docs/decisions/ADR-012-github-repo-as-sync-backend.md` | User-owned GitHub repo as durable sync backend |
| `docs/decisions/README.md` | ADR index updated with ADR-011 and ADR-012 |

## Child issues created

- #81 — Sub-Issue F: Netlify Functions for OAuth and GitHub API Proxy (no deps)
- #82 — Sub-Issue A: GitHub OAuth Login / Logout (no deps)
- #83 — Sub-Issue B: Repo Configuration (depends on #82)
- #84 — Sub-Issue E: Settings UI for Repo Config and Cache Parameter (depends on #82, #83)
- #85 — Sub-Issue C: Read Preferences from Remote Repo on Load (depends on #82, #83)
- #86 — Sub-Issue D: Write Preferences to Remote Repo on Update (depends on #82, #83, #85)

## Test plan

- [x] Review `business-specifications.md` — rules, edge cases, and testing guides are complete for all 6 sub-issues
- [x] Review `ADR-011` — OAuth flow, cookie attributes, credentials setup are accurate
- [x] Review `ADR-012` — JSON structure, write flow (including first-write / no-SHA case), cache strategy are correct
- [ ] Confirm `test-cases.md` covers all rules and edge cases
- [ ] Confirm all 6 child issues (#81–#86) have correct dependency references

Closes #64

### CODE REVIEW

File: docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+## Sub-Issue A — GitHub OAuth Login / Logout
+
+**Depends on:** nothing
+
+### Rules
+
+1. The user can initiate a GitHub OAuth login flow from the Settings page. On success, an HTTP-only cookie containing the access token is set by the Netlify function that handles the OAuth callback.
+2. The login flow uses the GitHub OAuth App authorization URL. The app requests only the minimum scopes needed to read and write a single file in one user-owned repository (`repo` scope or `public_repo` if the target repo is always public — see sub-issue B).
+3. After the callback, the user is redirected back to the Settings page with a visible success or error state.
+4. The user can log out; logout clears the HTTP-only cookie and removes the stored repo config from the Settings UI (but not from IndexedDB — station data is kept).
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md]: logout is not removing the stored repo config, but only disconnect the webapp from the GitHub OAuth.

File: docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+## Sub-Issue B — Repo Configuration
+
+**Depends on:** Sub-Issue A (user must be authenticated)
+
+### Rules
+
+1. The Settings page exposes two text inputs: `owner/repo` (e.g. `alice/my-stations`) and file path (e.g. `stations.json`). Both fields are required before any sync can occur.
+2. Configuration is validated on save: the Netlify proxy must be able to confirm the file path exists or the repo is reachable. If not, a human-readable error is shown.
+3. The configuration (`owner/repo`, file path, and `revalidate-cache-days`) is stored both in IndexedDB (for local persistence across reloads) and in the user's remote JSON file under a `"settings"` section (see "Remote JSON file structure" below). For unauthenticated users, these fields are empty.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md]: > For unauthenticated users, these fields are empty.

should rather be : 

> For never authenticated users, these fields are empty.

an unauthenticated user but who was authenticated previously will have the settings prefilled in IndexedDb.

File: docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+## Sub-Issue C — Read Preferences from Remote Repo on Load
+
+**Depends on:** Sub-Issues A, B
+
+### Rules
+
+1. On application load, if the user is authenticated and repo config is present, the app compares the age of the IndexedDB data against the configurable `revalidate-cache-days` parameter (default: 7 days, editable from Settings UI).
+2. If IndexedDB data is younger than the threshold, the remote repo is not consulted; IndexedDB is used as-is.
+3. If IndexedDB data is older than the threshold (or absent), the app fetches the JSON file from the remote repo via a Netlify proxy function and merges it into IndexedDB, replacing the existing data. The JSON file includes a `"settings"` section (see "Remote JSON file structure" below); the app reads `owner`, `repo`, and `revalidateCacheDays` from it and updates IndexedDB accordingly.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md]: > The JSON file includes a `"settings"` section (see "Remote JSON file structure" below); the app reads `owner`, `repo`, and `revalidateCacheDays` from it and updates IndexedDB accordingly

This is wrong: the  `owner`, `repo`, and `revalidateCacheDays` are set manually from the Settings page. 

The JSON file only includes the stations list and preferred fuel type

File: docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+**Depends on:** Sub-Issues A, B, C
+
+### Rules
+
+1. Whenever the user saves a change to their station list or fuel type default, the updated data is first written to IndexedDB, then a write request is sent to the remote repo via Netlify function.
+2. Before writing, the current file content is fetched from the remote repo and a diff is presented to the user (reusing the diff UI implemented in issue #63). The user must confirm before the write is committed.
+3. The write uses the GitHub Contents API `PUT /repos/{owner}/{repo}/contents/{path}` with the file's current `sha` to avoid overwrite conflicts.
+4. If the remote write fails, IndexedDB retains the update and a non-blocking error is shown; the write is not retried automatically.
+5. The JSON written to the remote repo always includes the `"settings"` section (see "Remote JSON file structure" below), preserving the current `owner`, `repo`, and `revalidateCacheDays` values alongside `stations` and `defaultFuel`.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md]: no need to save  `owner`, `repo`, and `revalidateCacheDays` values into the JSON file

File: docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+## Sub-Issue E — Settings UI for Repo Config and Cache Parameter
+
+**Depends on:** Sub-Issues A, B
+
+### Rules
+
+1. The Settings page gains a "GitHub Sync" section with: login/logout control, `owner/repo` field, file path field, and a `revalidate-cache-days` number input.
+2. `revalidate-cache-days` accepts a positive integer; values ≤ 0 are rejected with an inline validation message.
+3. All fields in this section are disabled when the user is not authenticated.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md]: to remove since:

- user who never authenticated needs to input  `owner`, `repo`, and `revalidateCacheDays` values
- user who previous authenticated may want to change  `owner`, `repo`, and `revalidateCacheDays` values

File: docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+### How to test locally
+
+1. Start the app with `netlify dev`.
+2. Navigate to the Settings page while unauthenticated; confirm all GitHub Sync fields (`owner/repo`, file path, `revalidate-cache-days`) are disabled/greyed out.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md]: to modify following comment on R153

File: docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+2. **Remote repository as user-data backend via GitHub Contents API** — The app currently stores all user data client-side only (ADR-008). Using a user-owned GitHub repo as a durable sync target is a new persistence tier that interacts with the existing IndexedDB layer and requires a cache-invalidation strategy. This warrants a dedicated ADR.
+
+## Remote JSON File Structure
+
+The JSON file stored in the user's GitHub repository has the following shape. For users who have never authenticated, the `settings` fields are empty/absent locally; for authenticated users, the settings are persisted both in IndexedDB and in the remote file:
+
+```json
+{
+  "settings": {
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md]: to remove

File: docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+  },
+  "stations": [
+    { "name": "Station A", "url": "https://..." },
+    { "name": "Station B", "url": "https://..." }
+  ],
+  "defaultFuel": "SP95"
+}
+```
+
+- For unauthenticated users, the `settings` section is absent from any local representation; these fields are only populated once the user logs in and saves repo config.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/business-specifications.md]: to remove

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 9]
```diff
@@ -0,0 +1,225 @@
+# Test Cases — Issue #64: GitHub Repository Preferences
+
+## Sub-Issue A — GitHub OAuth Login / Logout
+
+### A-1: Login button visible when unauthenticated
+- **Precondition:** No `gh_token` cookie is set.
+- **Action:** Navigate to the Settings page.
+- **Expected:** A "Login with GitHub" button is visible. All GitHub Sync fields (`owner/repo`, file path, `revalidate-cache-days`) are disabled/greyed out.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: `owner/repo`, file path, `revalidate-cache-days` are not greyed as detailled in the business specs

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+
+### A-2: Login flow redirects to GitHub authorization page
+- **Precondition:** No `gh_token` cookie is set.
+- **Action:** Click "Login with GitHub" on the Settings page.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md:17]: `owner/repo`, file path, `revalidate-cache-days` must be set to enable the button

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+- **Expected:** The browser redirects to GitHub's OAuth authorization URL (`github.com/login/oauth/authorize`), including the `client_id`, `scope`, and `state` query parameters.
+
+### A-3: Successful OAuth callback sets cookie and redirects to Settings
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: will this be testable in vitest tests?

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+- **Action:** The `github-auth-callback` Netlify function receives the `code` and valid `state` parameter.
+- **Expected:** The function exchanges the code for a token, sets the `gh_token` cookie (`HttpOnly`, `SameSite=Strict`, `Max-Age=28800`), and redirects to `/settings?auth=success`. The Settings page shows an authenticated state.
+
+### A-4: Authenticated state persists across page reloads
+- **Precondition:** `gh_token` cookie is present and valid.
+- **Action:** Reload the Settings page.
+- **Expected:** The user remains authenticated. GitHub Sync fields are enabled.
+
+### A-5: Logout clears authentication state and disables fields
+- **Precondition:** User is authenticated (`gh_token` cookie present).
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: no disabling fields as stated in business specs

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+- **Action:** The `github-auth-callback` function receives the error parameter.
+- **Expected:** The function redirects to `/settings?auth=error`. The Settings page displays a human-readable error message. No cookie is set.
+
+### A-8: state parameter mismatch in OAuth callback is rejected
+- **Precondition:** The `state` parameter in the callback does not match the one generated at login start.
+- **Action:** The `github-auth-callback` function receives a mismatched `state`.
+- **Expected:** The function does not exchange the code for a token. The response is an error redirect (or equivalent failure). No cookie is set.
+
+### A-9: Token expiry detected on GitHub API call
+- **Precondition:** User has a `gh_token` cookie that has been revoked on GitHub's side (simulated by revoking in GitHub OAuth settings).
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md:53]: is this test a manual one?

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+- **Action:** User triggers any action that calls the GitHub API proxy.
+- **Expected:** The proxy receives a 401 from GitHub, clears the `gh_token` cookie, and the UI prompts the user to log in again.
+
+---
+
+## Sub-Issue B — Repo Configuration
+
+### B-1: owner/repo and file path inputs are enabled when authenticated
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: as I read this test, I think that the fields should be disabled if user is authenticated. and a message is displayed "If you need to change your owner, repository and file path details, you need to logout, update the values and reauthenticate."

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** User is authenticated.
+- **Action:** Navigate to the Settings page.
+- **Expected:** The `owner/repo` and file path inputs are enabled and editable.
+
+### B-2: Valid repo config saves successfully
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: to discard?

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** User is authenticated. Inputs contain a valid `owner/repo` and file path pointing to an accessible repository.
+- **Action:** Click Save.
+- **Expected:** A success message appears. No error is shown. The values are persisted in IndexedDB and reloaded on next page visit.
+
+### B-3: Saved config persists across reloads
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: to replace with a test checking persistance of values in IndexedDb?

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** User has saved a valid repo config.
+- **Action:** Reload the Settings page.
+- **Expected:** The `owner/repo` and file path fields display the previously saved values.
+
+### B-4: Invalid repo config shows human-readable error
+- **Precondition:** User is authenticated. Inputs contain a non-existent `owner/repo`.
+- **Action:** Click Save.
+- **Expected:** A human-readable error message is displayed. No success message appears.
+
+### B-5: Fields are empty and disabled after logout
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: same as A-1 and A-5 regarding disabling fields as unauthenticated.

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** User is authenticated and repo config is present. IndexedDB timestamp is older than `revalidate-cache-days`.
+- **Action:** Load the application.
+- **Expected:** The app fetches the remote JSON file via the Netlify proxy. IndexedDB is updated with the remote data. The timestamp is reset to the current date and time.
+
+### C-3: Remote repo not fetched again after fresh load
+- **Precondition:** The remote repo was just fetched (timestamp is now current).
+- **Action:** Reload the application immediately.
+- **Expected:** No request is made to the GitHub API proxy.
+
+### C-4: IndexedDB timestamp resets after user-triggered update
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: should have one test per action (add, edit, delete) and per data changed (a station and fuel type)

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** User is authenticated with a saved station list.
+- **Action:** Add, edit, or delete a station, or change the fuel type default.
+- **Expected:** The IndexedDB timestamp is updated to the current date and time.
+
+### C-5: settings section from remote file overrides local settings
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: not necessary anymore per business specs update in previous comments

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+### C-7: Remote fetch failure (401) clears cookie and shows warning
+- **Precondition:** User's access token has been revoked.
+- **Action:** App loads with a stale IndexedDB timestamp and the proxy returns 401.
+- **Expected:** The app prompts re-authentication. If the user refuses, the `gh_token` cookie is cleared and a warning banner states that GitHub access was revoked, noting that IndexedDB data is being used.
+
+---
+
+## Sub-Issue D — Write Preferences to Remote Repo on Update
+
+### D-1: Diff dialog shown before write
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: one test per action (add, edit)

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 21]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** User is authenticated and repo config is present. Remote file exists.
+- **Action:** Add or edit a station.
+- **Expected:** A diff dialog appears showing the before (remote) and after (updated) JSON content. The write is not performed until the user confirms.
+
+### D-2: Confirmed write updates the remote file
+- **Precondition:** Diff dialog is shown.
+- **Action:** User confirms the diff dialog.
+- **Expected:** The GitHub Contents API `PUT` is called with the correct `sha`. The remote file is updated. A success indicator appears.
+
+### D-3: Cancelled write leaves remote file unchanged, IndexedDB updated
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: user must be informed that local data (IndexedDb) differs from remote data. do the specs cover that?

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 22]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** Diff dialog is shown.
+- **Action:** User cancels the diff dialog.
+- **Expected:** No write is sent to the remote repo. IndexedDB retains the local update.
+
+### D-4: Write to non-existent remote file shows empty diff baseline
+- **Precondition:** User is authenticated and repo config is present. No remote file exists yet.
+- **Action:** Add a station (triggering the first write).
+- **Expected:** The diff dialog shows an empty baseline (no existing content) versus the new JSON. On confirm, the file is created in the remote repo.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: no diff dialog if file doesn't exist

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 23]
```diff
@@ -0,0 +1,225 @@
...
+### D-7: Written JSON always includes settings section
+- **Precondition:** User confirms a write after adding a station.
+- **Action:** Inspect the JSON written to the remote repo.
+- **Expected:** The file contains `settings`, `stations`, and `defaultFuel` keys. The `settings` section reflects the current `owner`, `repo`, and `revalidateCacheDays` values.
+
+---
+
+## Sub-Issue E — Settings UI for Repo Config and Cache Parameter
+
+### E-1: revalidate-cache-days field accepts valid positive integer
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md:188]: In previous comments, I may have said the `revalidate-cache-days` is not editable while authenticated. after a deeper thought that overwrites previous instructions about this field, I think it is the only value always editable regardless of user's oauth status

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 24]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** User is authenticated.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: should be none based on previous comment R161

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 25]
```diff
@@ -0,0 +1,225 @@
...
+- **Action:** Enter `7` in the `revalidate-cache-days` input and save.
+- **Expected:** The value saves successfully. No validation error is shown.
+
+### E-2: revalidate-cache-days field rejects zero
+- **Precondition:** User is authenticated.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: should be none based on previous comment R161

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 26]
```diff
@@ -0,0 +1,225 @@
...
+- **Action:** Enter `0` in the `revalidate-cache-days` input and attempt to save.
+- **Expected:** An inline validation error appears. The form cannot be submitted.
+
+### E-3: revalidate-cache-days field rejects negative numbers
+- **Precondition:** User is authenticated.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: should be none based on previous comment R161

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 27]
```diff
@@ -0,0 +1,225 @@
...
+- **Action:** Enter `-3` in the `revalidate-cache-days` input and attempt to save.
+- **Expected:** An inline validation error appears. The form cannot be submitted.
+
+### E-4: All GitHub Sync fields disabled when unauthenticated
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: per business specs updates in previous comments, test should read "All GitHub Sync fields enabled when unauthenticated"

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 28]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** No `gh_token` cookie is set.
+- **Action:** Navigate to the Settings page.
+- **Expected:** The `owner/repo`, file path, and `revalidate-cache-days` fields are all disabled/greyed out.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: per previous comment R176, expected needs to be updated

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 29]
```diff
@@ -0,0 +1,225 @@
...
+
+### E-5: All GitHub Sync fields enabled after login
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: per business specs updates in previous comments, test should read "Owner and Repo GitHub Sync fields disabled after login"

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 30]
```diff
@@ -0,0 +1,225 @@
...
+- **Precondition:** User completes the OAuth login flow.
+- **Action:** Navigate to (or reload) the Settings page.
+- **Expected:** All three fields (`owner/repo`, file path, `revalidate-cache-days`) are enabled and editable.
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: per previous comment R181, expected needs to be updated

File: docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md [hunk 31]
```diff
@@ -0,0 +1,225 @@
...
+
+### E-6: All GitHub Sync fields disabled after logout
```
Comment (JeremieLitzler) [inline docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md]: similar updates needed as test E-4

File: docs/decisions/ADR-011-github-oauth-app-auth.md [hunk 32]
```diff
@@ -0,0 +1,103 @@
...
+Use a **GitHub OAuth App** with the **Authorization Code flow**, exchanging the code **server-side** inside a Netlify function to keep the Client Secret off the browser. Store the resulting access token in an **HTTP-only, SameSite=Strict, Secure cookie** with a lifetime of **8 hours** (`Max-Age=28800`).
+
+Two Netlify functions handle the flow:
+
+- `github-auth-start` — generates a random `state` value (stored in a short-lived cookie for CSRF protection), then redirects the browser to GitHub's OAuth authorization URL with the correct `client_id`, `scope`, and `state`.
+- `github-auth-callback` — receives the `code` and `state` from GitHub's redirect, validates `state`, exchanges `code` for an access token via a server-to-server POST to `https://github.com/login/oauth/access_token`, sets the HTTP-only cookie, then redirects the browser to `/settings?auth=success` (or `/settings?auth=error` on failure).
+
+The SPA detects auth state by calling the GitHub API proxy (ADR-012) and checking whether it returns a valid response or a 401.
+
+Logout is implemented by a Netlify function that clears the cookie.
```
Comment (JeremieLitzler) [inline docs/decisions/ADR-011-github-oauth-app-auth.md:34]: no tests presents for this function in @docs/prompts/tasks/issue-64-github-repo-preferences/test-cases.md

File: docs/decisions/ADR-012-github-repo-as-sync-backend.md [hunk 33]
```diff
@@ -0,0 +1,108 @@
+# ADR-012: User-Owned GitHub Repository as Remote Sync Backend
+
+**Date:** 2026-04-30
+**Status:** Accepted
+
+## Context
+
+Issue #64 introduces preference sync: stations, default fuel type, and repository configuration should persist beyond a single device and survive browser storage clearing.
```
Comment (JeremieLitzler) [inline docs/decisions/ADR-012-github-repo-as-sync-backend.md]: as per business specs comments, no more "repository configuration" in the file

File: docs/decisions/ADR-012-github-repo-as-sync-backend.md [hunk 34]
```diff
@@ -0,0 +1,108 @@
...
+
+Use a **user-owned GitHub repository** as the remote sync backend. A single JSON file (path configured by the user) is read on app load and written on preference change via the **GitHub Contents API** (`GET /repos/{owner}/{repo}/contents/{path}` and `PUT /repos/{owner}/{repo}/contents/{path}`).
+
+All GitHub API calls are routed through a Netlify function (`github-api-proxy`) that reads the HTTP-only access token cookie (set by ADR-011) and forwards the request. The browser never holds the token.
+
+### Remote JSON File Structure
+
+```json
+{
+  "settings": {
```
Comment (JeremieLitzler) [inline docs/decisions/ADR-012-github-repo-as-sync-backend.md]: section to remove

File: docs/decisions/ADR-012-github-repo-as-sync-backend.md [hunk 35]
```diff
@@ -0,0 +1,108 @@
...
+    "revalidateCacheDays": 7
+  },
+  "stations": [
+    { "name": "Station name", "url": "https://..." }
+  ],
+  "defaultFuel": "SP95"
+}
+```
+
+For unauthenticated users the `settings` section is absent from IndexedDB and the remote file does not exist. Fields are only populated and persisted after the user authenticates and saves their repo configuration.
```
Comment (JeremieLitzler) [inline docs/decisions/ADR-012-github-repo-as-sync-backend.md]: to remove

File: docs/decisions/ADR-012-github-repo-as-sync-backend.md [hunk 36]
```diff
@@ -0,0 +1,108 @@
...
+
+### Cache Invalidation Strategy
+
+To avoid a GitHub API call on every page load, a timestamp is stored in IndexedDB alongside the data. On app load:
+
+1. Compare `Date.now()` against the stored timestamp.
```
Comment (JeremieLitzler) [inline docs/decisions/ADR-012-github-repo-as-sync-backend.md]: unit test must mock `Date.now()`

File: docs/decisions/ADR-012-github-repo-as-sync-backend.md [hunk 37]
```diff
@@ -0,0 +1,108 @@
...
+- ⚠️ 409 conflict on concurrent writes from multiple devices requires manual intervention (no automatic merge)
+- ⚠️ User must create and configure the repository and file path manually before sync works
+- ⚠️ First-time setup requires the user to understand what `owner/repo` means — mitigated by UI guidance
+- ⚠️ Platform dependency on GitHub's API availability — app degrades gracefully to local-only when the API is unreachable
+
+## Alternatives Considered
+
+1. **Netlify Blobs**: Platform-specific, not user-owned, would require tying storage to the Netlify site identity — rejected.
+2. **Dedicated Express/Node backend + PostgreSQL**: Correct at scale but overkill for a single-user preferences file — rejected.
+3. **Firebase / Supabase**: Third-party dependency with their own auth and pricing — rejected in favour of the user's existing GitHub account.
```
Comment (JeremieLitzler) [inline docs/decisions/ADR-012-github-repo-as-sync-backend.md]: Remove Firebase mention completely.

Regarding Supabase, note that it might be a good idea if the app becomes a profitable service.

**Review by JeremieLitzler — COMMENTED** [review]
all comments to be addressed before coding
